src/hooks/useCKBNode.tsx

```diff
@@ -24,15 +48,59 @@ interface CKBNodeProviderProps {
 }
 
 export const CKBNodeProvider = ({ children, defaultEndpoint }: PropsWithChildren<CKBNodeProviderProps>) => {
-  const nodeService = new NodeService(defaultEndpoint)
+  const [nodeService, setNodeService] = useState(new NodeService(defaultEndpoint))
+  const [nodes, _setNodes] = useState<CKBNode[]>(loadEndpoints())
```

Comment:
The pair of `nodes` and `setNodes` is conventional, it's better to have another name at line 60